src/FffGcodeWriter.cpp

```diff
@@ -1592,6 +1547,62 @@ void FffGcodeWriter::processSkinInsets(const SliceDataStorage& storage, LayerPla
+{
+    const coord_t skin_line_width = mesh_config.skin_config.getLineWidth();
+    const int top_bottom_extruder_nr = mesh.getSettingAsExtruderNr("top_bottom_extruder_nr");
+
+    Polygons skin_polygons;
+    Polygons skin_lines;
+
+    const bool process_skin_infill = extruder_nr == top_bottom_extruder_nr;
+
+    // generate skin_polygons and skin_lines (and concentric_perimeter_gaps if needed)
+    if (process_skin_infill || generate_perimeter_gaps)
+    {
+        // only generate infill if we're going to print it with this extruder
+        // or when we need to compute the perimeter gaps
+        // TODO: only compute this once when the infill is concentric and the perimeter gpas are printed with a different extruder
+
+        // calculate bridging angle
+        int bridge = -1;
+        if (layer_nr > 0)
+        {
+            bridge = bridgeAngle(skin_part.outline, &mesh.layers[layer_nr-1]);
+        }
+        if (bridge > -1)
+        {
+            pattern = EFillMethod::LINES; // force lines pattern when bridging
+            skin_angle = bridge;
+        }
+
+        // calculate polygons and lines
+        int extra_infill_shift = 0;
+        coord_t offset_from_inner_skin_infill = 0;
+        Polygons* perimeter_gaps_output = (generate_perimeter_gaps)? &concentric_perimeter_gaps : nullptr;
+        Infill infill_comp(pattern, skin_part.inner_infill, offset_from_inner_skin_infill, skin_line_width, skin_line_width, skin_overlap, skin_angle, z, extra_infill_shift, perimeter_gaps_output);
+        infill_comp.generate(skin_polygons, skin_lines);
+    }
+
+    // add skin itself!
+    if (process_skin_infill && (skin_polygons.size() > 0 || skin_lines.size() > 0))
+    {
+        added_something = true;
+        setExtruder_addPrime(storage, gcode_layer, layer_nr, extruder_nr);
+        gcode_layer.setIsInside(true); // going to print stuff inside print object
+        gcode_layer.addPolygonsByOptimizer(skin_polygons, &mesh_config.skin_config);
+
+        if (pattern == EFillMethod::GRID || pattern == EFillMethod::LINES || pattern == EFillMethod::TRIANGLES || pattern == EFillMethod::CUBIC || pattern == EFillMethod::TETRAHEDRAL || pattern == EFillMethod::CUBICSUBDIV)
+        {
+            gcode_layer.addLinesByOptimizer(skin_lines, &mesh_config.skin_config, SpaceFillType::Lines, mesh.getSettingInMicrons("infill_wipe_dist"));
+        }
+        else
+        {
+            gcode_layer.addLinesByOptimizer(skin_lines, &mesh_config.skin_config, (pattern == EFillMethod::ZIG_ZAG)? SpaceFillType::PolyLines : SpaceFillType::Lines);
+        }
+    }
```

Comment:
The generating of lines for the perimeter gaps isn't in this function. Shouldn't that be close to the generating of polygons?